### PR TITLE
Slombard/icp 19 add images to the memories

### PR DESCRIPTION
**Pull request by 552020**
Open this only for visibility. It deploy but it's not tested. Not sure how to test. I tried to write a Python script but it's getting to complicated.

### CODE REVIEW

**Comment by 552020** [comment]
This could be reviewed now @dantol29 @lmangall 

There is bash test in the tests folder: test_image_new.sh is working.

File: tests/test_image.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import base64
+import subprocess
+from ic.candid import Types, encode
+
+# Paths for test image and output
+image_path = "Futura.png"
+output_image_path = "output_image.png"
+
```
Comment (dantol29) [inline tests/test_image.py]: Do we still need python tests

File: tests/test_image.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+IMAGE_PATH="Futura.png"
+OUTPUT_IMAGE_PATH="output_image.png"
+CANDID_LOG_FILE="test_image_candid.log"
+ERROR_LOG_FILE="test_image_error.log"
+CANISTER_ID="a4tbr-q4aaa-aaaaa-qaafq-cai"
```
Comment (dantol29) [inline tests/test_image.sh]: canister id should be fetched from .env:
```
source ../.env
echo "canister: $CANISTER_ID_FUTURA_BACKEND"

IMAGE_PATH="Futura.png"
OUTPUT_IMAGE_PATH="output_image.png"
CANDID_LOG_FILE="test_image_candid.log"
ERROR_LOG_FILE="test_image_error.log"
CANISTER_ID=$CANISTER_ID_FUTURA_BACKEND
```

File: tests/test_image.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+echo "" > "$CANDID_LOG_FILE"
+echo "" > "$ERROR_LOG_FILE"
+
+# Read the image as raw binary and encode it as a hex string
+# IMAGE_BINARY=$(xxd -p "$IMAGE_PATH" | tr -d '\n')
+IMAGE_BINARY="89504e470d"  # This is a part of a PNG file header
+
```
Comment (dantol29) [inline tests/test_image.sh]: Can you remove outdated tests?

File: src/Futura_backend/src/lib.rs [hunk 4]
```diff
@@ -1,43 +1,106 @@
...
+}
+
+/* TODO: Consider size limitations for large image data. Since images can be large,
+     we may need to implement chunking or compression to ensure the serialized data
+     fits within the limits of stable memory.
+*/
 
+// Serialize and store the state before upgrading
+#[pre_upgrade]
+fn pre_upgrade() {
```
Comment (dantol29) [inline src/Futura_backend/src/lib.rs:55]: I had to comment out pre_upgrade and post_upgrade to deploy

**Review by dantol29 — CHANGES_REQUESTED** [review]

File: tests/test_image.py (hunk 1 above)
Comment (552020) [reply tests/test_image.py]: No! I mean maybe.

File: tests/test_image.sh (hunk 2 above)
Comment (552020) [reply tests/test_image.sh]: Ok!

File: tests/test_image.sh (hunk 3 above)
Comment (552020) [reply tests/test_image.sh]: Yes!

File: src/Futura_backend/src/lib.rs (hunk 4 above)
Comment (552020) [reply src/Futura_backend/src/lib.rs:55]: Why? Which errors did you get?

**Comment by dantol29** [comment]
Everything works!

**Pull request merged**